pkg/rbd/nodeserver.go

```diff
@@ -70,10 +70,18 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	if !notMnt {
 		return &csi.NodePublishVolumeResponse{}, nil
 	}
-	volOptions, err := getRBDVolumeOptions(req.GetVolumeContext())
+
+	ignoreMultiWriterEnabled := true
+	if req.VolumeCapability.AccessMode.Mode != csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER {
```

Comment:
The only valid modes that can even make it here from the CO are the ones we advertise in the capabilities (SINGLE_NODE_WRITER and MULTI_NODE_MULTI_WRITER), anything else will be rejected by the CO because it's not in the capabilities.  The next step here was to add the additional capabilities and continue handling cases as needed (those that weren't sub/super sets).  In this case you have one or the other, so the `!=`  works.

Reply:
Well the suggested comparison expresses your intent and the actual comparison only works accidentally, as long as we continue to only advertise those two capabilities.  I can't see any reason why we wouldn't decide in the future to also advertise other capabilities -- e.g. SINGLE_NODE_MULTI_WRITER or the read-only capabilities.

Reply:
This is all really rather irrelevant, you're reviewing a merged PR :)  In addition, based on your objections, this PR is being reverted and replaced.  Let's see how the new PR goes and just save the discussions for that.